### PR TITLE
Handle pandas DataFrames without index 0

### PR DESCRIPTION
**Pull request by AlexandreGuinaudeau**
We're currently hitting a KeyError if the pandas DataFrame does not have any row with index "0"

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -21,7 +21,7 @@ pandas = "^1.4.1"
 pytest = "^7.0.1"
 pytest-mockito = "^0.0.4"
 tox = "^3.24.5"
-mypy = "^0.942"
+mypy = "^0.981"
```
Comment (AlexandreGuinaudeau) [inline pyproject.toml:24]: Fixes https://github.com/python/mypy/issues/13627

File: tox.ini [hunk 2]
```diff
@@ -3,25 +3,25 @@ isolated_build = true
 envlist = py38, py39, py310, pylint, mypy, black
 
 [testenv]
-whitelist_externals = poetry
+allowlist_externals = poetry
```
Comment (AlexandreGuinaudeau) [inline tox.ini:6]: breaking change in poetry

File: tox.ini [hunk 3]
```diff
@@ -3,25 +3,25 @@ isolated_build = true
...
 commands =
-    poetry install
+    poetry install -q
```
Comment (AlexandreGuinaudeau) [inline tox.ini:8]: poetry install was failing to run in CircleCI with the interactive output showing download progress

File: .circleci/config.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ version: 2
 jobs:
   py38:
     docker:
-      - image: cimg/python:3.8.12
+      - image: cimg/python:3.8.16
```
Comment (AlexandreGuinaudeau) [inline .circleci/config.yml:13]: There was a breaking change in pypi which made it incompatible with the version of Poetry pre-installed in the previous image versions

**Review by ahggns — APPROVED** [review]

**Pull request merged**